integreat_cms/api/decorators.py

```diff
@@ -117,3 +125,74 @@ def wrap(request, *args, **kwargs):
             return JsonResponse({"error": str(e) or "Not found."}, status=404)
 
     return wrap
+
+
+def matomo_tracking(func):
+    """
+    This decorator is supposed to be applied to API content endpoints. It will track
+    the request in Matomo. The request to the Matomo API is executed asynchronously in its
+    own thread to not block the Integreat CMS API request.
+
+    Only the URL and the User Agent will be sent to Matomo.
+
+    :param func: decorated function
+    :type func: ~collections.abc.Callable
+
+    :return: The decorated feedback view function
+    :rtype: ~collections.abc.Callable
+    """
+
+    @wraps(func)
+    def wrap(request, *args, **kwargs):
+        r"""
+        The inner function for this decorator.
+
+        :param request: Django request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied kwargs
+        :type \**kwargs: dict
+
+        :return: The response of the given function or an 404 :class:`~django.http.JsonResponse`
+        :rtype: ~django.http.JsonResponse
+        """
+        data = {
+            "idsite": request.region.matomo_id,
+            "token_auth": request.region.matomo_token,
+            "rec": 1,
+            "url": request.build_absolute_uri(),
+            "urlref": settings.BASE_URL,
+            "ua": request.META["HTTP_USER_AGENT"]
+            if "HTTP_USER_AGENT" in request.META
+            else "unknown user agent",
+            "cip": f"{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}",
+        }
+
+        t = threading.Thread(target=matomo_request, args=(settings.MATOMO_URL, data))
+        t.setDaemon(True)
+        t.start()
+        return func(request, *args, **kwargs)
+
+    def matomo_request(host, data):
+        """
+        Wrap HTTP request to Matomo in threaded function.
+
+        :param host: Hostname + protocol of Matomo installation
+        :type host:
+
+        :param data: Data to send to Matomo API
+        :type data: dict
+        """
+        data = parse.urlencode(data)
+        url = f"{host}/matomo.php?{data}"
+        req = request.Request(url)
+        try:
+            with request.urlopen(req):
+                pass
+        except error.HTTPError as e:
+            logger.error("Matomo Tracking API request failed with: %s", e)
```

Comment:
```suggestion
            logger.error(
                "Matomo Tracking API request to %r failed with: %s",
                # Mask auth token in log
                re.sub(r"&token_auth=[^&]+", "&token_auth=********", url),
                e,
            )
```
Maybe it could be helpful to know which request failed exactly?

Reply:
Indeed. I had the same though already as well.